numpy/core/tests/test_longdouble.py

```diff
@@ -10,29 +10,31 @@
 from numpy.compat import sixu
 from test_print import in_foreign_locale
 
-longdouble_longer_than_double = (np.finfo(np.longdouble).eps
-                                 < np.finfo(np.double).eps)
+LD_INFO = np.finfo(np.longdouble)
+longdouble_longer_than_double = (LD_INFO.eps < np.finfo(np.double).eps)
 
 
-_o = 1 + np.finfo(np.longdouble).eps
+_o = 1 + LD_INFO.eps
 string_to_longdouble_inaccurate = (_o != np.longdouble(repr(_o)))
 del _o
 
 
 def test_scalar_extraction():
     """Confirm that extracting a value doesn't convert to python float"""
-    o = 1 + np.finfo(np.longdouble).eps
+    o = 1 + LD_INFO.eps
     a = np.array([o, o, o])
     assert_equal(a[1], o)
 
 
 # Conversions string -> long double
 
 
+@dec.skipif(LD_INFO.precision >= len(repr(np.longdouble(0.1))) - 2,
```

Comment:
Probably better computing `len(repr(np.longdouble(0.1)))` on another line, doing `LD_INFO.precision + 2 >= repr_precision`, and adding a comment linking that `+2` to `scalartypes`